server/src/http/feature_refresher.rs

```diff
@@ -172,14 +177,14 @@ impl FeatureRefresher {
         }
     }
 
-    fn get_filtered_features(&self, token: &EdgeToken) -> Option<ClientFeatures> {
+    fn get_features_by_filter(
+        &self,
+        token: &EdgeToken,
+        filters: FeatureFilterSet,
+    ) -> Option<ClientFeatures> {
         self.features_cache
             .get(&cache_key(token))
-            .map(|e| e.value().clone())
-            .map(|features_response| ClientFeatures {
-                features: features_response.features.filter_by_projects(token),
-                ..features_response
-            })
+            .map(|client_features| filter_client_features(&client_features, filters))
```

Comment:
Ok, I think the answer to my comment above lies in this method. We're cloning the list inside. I'm just wondering if we could return a reference to the features instead of a clone, and then let the client create the clone if they need to.

The current way is a bit safer, but if we're going to go down the optimization path might be something to consider (although not for this PR)

Reply:
Probably not sadly, this would require us to marry the lifetime of the DashMap data to the lifetime of the features we want to return. Because our dashmap is currently a self reference that lifetime is already has some hefty constraints on it

I don't think this matters that much, though, the data in dashmap is borrowed by nature and the data we require to send over HTTP needs to be an owned reference so at least one clone is absolutely necessary